lib/system.nim

```diff
@@ -1522,10 +1522,12 @@ include "system/iterators_1"
 
 {.push stackTrace: off.}
 
-proc abs*(x: float64): float64 {.noSideEffect, inline.} =
-  if x < 0.0: -x else: x
-proc abs*(x: float32): float32 {.noSideEffect, inline.} =
-  if x < 0.0: -x else: x
+proc abs*[T: float64 | float32](x: T): T {.noSideEffect, inline.} =
+  if x < 0.0: -x
+  elif x > 0.0: x
```

Comment:
The ``elif x > 0.0: x`` case can be merged with the final ``else: x``. I think. If not, add a comment please.

Reply:
Yeah, it can't handle `NaN`, I will add a comment.

Reply:
This is how this PR looked before, and it introduced a heavy performance regression.
This extra branching seemed to help the predictor somewhat, but this benchmark shows that we are really just playing games with the branch predictor:
```nim
proc abs1(x: float64): float64 {.noSideEffect, noinline.} =
  if x < 0.0: -x
  elif x > 0.0: x
  elif x == 0.0: 0.0
  else: x

proc abs2(x: float64): float64 {.noSideEffect, noinline.} =
  if x < 0.0: -x
  elif x == 0.0: 0.0
  else: x

proc fabs(x: cdouble): cdouble {.importc.}

proc abs3(x: float64): float64 {.noSideEffect, noinline.} =
  fabs(x)

proc abs4(x: float64): float64 {.noSideEffect, noinline.} =
  if (cast[uint64](x) and 1'u64 shl 63) != 0: -x
  else: x

proc abs5(x: float64): float64 {.noSideEffect, noinline.} =
  cast[float64](cast[uint64](x) and not(1'u64 shl 63))

import bitops
proc abs6[T: float64](x: T): T {.noSideEffect, inline.} =
  var x = cast[uint64](x)
  clearBit(x, 1)
  result = cast[float64](x)

import times
import random

template work(algo) =
  let n = 1000_000_000
  let t = cpuTime()
  type T = float64
  var tot = T(0.0)
  for i in 0..<n:
    let x = T(i*7) - tot
    tot += algo(x)
    tot += algo(-x)
  let t2 = cpuTime()
  echo (astToStr(algo), t2-t, tot)

proc main()=
  work(abs1)
  work(abs2)
  work(abs3)
  work(abs4)
  work(abs5)
main()
```
depending on how big you set `n`, the branch versions start to win or loose against the bit twiddling version or `fabs` depending on your compiler/arch.